cli/src/pcluster/validators/s3_validators.py

```diff
@@ -64,3 +66,62 @@ def _validate_s3_uri(self, s3_url):
                 ),
                 FailureLevel.ERROR,
             )
+
+
+class S3BucketUriValidator(Validator):
+    """S3 Bucket Url Validator."""
+
+    def _validate(self, url):
+
+        if urlparse(url).scheme == "s3":
+            try:
+                bucket = get_bucket_name_from_s3_url(url)
+                S3Client().head_bucket(bucket_name=bucket)
+            except ClientError as client_error:
+                message = _process_generic_s3_bucket_error(client_error, bucket)
+                self._add_failure(message, FailureLevel.ERROR)
+        else:
+            self._add_failure(f"The value '{url}' is not a valid S3 URI.", FailureLevel.ERROR)
+
+
+class S3BucketValidator(Validator):
+    """S3 Bucket Validator."""
+
+    def _validate(self, bucket):
+
+        s3_client = S3Client()
```

Comment:
You should use AWSApi().instance().s3 rather than calling the client directly.

Reply:
Done